File: CampusConnect.Server/Models/Dtos/LoginResponseDto.cs
namespace CampusConnect.Server.Models.Dtos
{
    public class LoginResponseDto
    {
        public string Token { get; set; }
        public string Username { get; set; }
        public UserRole Role { get; set; }
        public bool RequiresTwoFactor { get; set; } = false;
        public string? TempToken { get; set; }
    }
}

File: campusconnect.client/src/app/entities/loginResponseDto.ts
import { UserRoleEntity } from "./userRoleEntity";

export class LoginResponseDto {
    constructor(
        token: string,
        username: string,
        role: UserRoleEntity,
        requiresTwoFactor?: boolean,
        tempToken?: string
    ) {
        this.role = role;
        this.username = username;
        this.token = token;
        this.requiresTwoFactor = requiresTwoFactor || false;
        this.tempToken = tempToken;
    }

    public token: string;
    public username: string;
    public role: UserRoleEntity;
    public requiresTwoFactor: boolean;
    public tempToken?: string;
}